src/ucp/core/ucp_listener.c

```diff
@@ -161,56 +175,68 @@ ucp_listen(ucp_listener_h listener, const ucp_listener_params_t *params)
                                        UCS_SOCKADDR_STRING_LEN),
                       ucs_status_string(status));
 
-            if (status == UCS_ERR_BUSY) {
-                goto err_destroy_listeners;
+            if (status != UCS_ERR_BUSY) {
```

Comment:
maybe flatten the code:
if (status == UCS_OK) {
 ucs_debug(..)
} else if (status == BUSY && use_any_port) {
  /* retry another port */
} else {
  ucs_error
}